src/unix/stream.c

```diff
@@ -1243,8 +1243,9 @@ int uv_shutdown(uv_shutdown_t* req, uv_stream_t* stream, uv_shutdown_cb cb) {
   stream->flags |= UV_HANDLE_SHUTTING;
   stream->flags &= ~UV_HANDLE_WRITABLE;
 
-  uv__io_start(stream->loop, &stream->io_watcher, POLLOUT);
-  uv__stream_osx_interrupt_select(stream);
+  if (QUEUE_EMPTY(&stream->write_queue)) {
+    uv__io_feed(stream->loop, &stream->io_watcher);
+  }
```

Comment:
```suggestion
  if (QUEUE_EMPTY(&stream->write_queue))
    uv__io_feed(stream->loop, &stream->io_watcher);
```
Suggestion: update the comment on line 1238 to explain the shutdown(2) call is deferred to `uv__drain()`.